src/lfortran/semantics/ast_common_visitor.h

```
@@ -8944,13 +8945,19 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
 
         SymbolTable *tu_symtab = ASRUtils::get_tu_symtab(current_scope);
 
-        ASR::Module_t *m = ASRUtils::load_module(al, tu_symtab, module_name,
-                loc, true, compiler_options.po, true,
-                [&](const std::string &msg, const Location &loc) {
-                        diag.add(Diagnostic(msg, Level::Error, Stage::Semantic, {Label("", {loc})}));
-                        throw SemanticAbort();
-                    }, lm, compiler_options.generate_object_code
-                );
+        ASR::Module_t *m = nullptr;
+        try {
+            m = ASRUtils::load_module(al, tu_symtab, module_name,
+                    loc, true, compiler_options.po, true,
+                    [&](const std::string &msg, const Location &loc) {
+                            diag.add(Diagnostic(msg, Level::Error, Stage::Semantic, {Label("", {loc})}));
+                            throw SemanticAbort();
+                        }, lm, compiler_options.separate_compilation
+                    );
+        } catch ( LCompilersException &e ) {
+            std::cerr << e.name() + ": " << e.msg() << std::endl;
+            std::abort();
+        }
```

This approach may not be the best, because it will catch any possible ICE (if you go inside `load_module` there are other possible places where ICE is raised, we wouldn't want to change that for everything) and throw it now as a `std::cerr`, which isn't what we want.

This is the place where ICE is being throw from currently: https://github.com/lfortran/lfortran/blob/aba1a616dccc299442489745254333f291df758e/src/libasr/modfile.cpp#L149-L151

maybe throw an error there itself, would it work?

---

If you think that will work, maybe then you can also change the title of the PR, to something like: `fix: raise standard error when incompatible mod file is used`?